server/datastore/mysql/software_installers.go

```diff
@@ -174,8 +174,9 @@ func (ds *Datastore) getOrGenerateSoftwareInstallerTitleID(ctx context.Context,
 	insertArgs := []any{payload.Title, payload.Source}
 
 	if payload.BundleIdentifier != "" {
-		selectStmt = `SELECT id FROM software_titles WHERE bundle_identifier = ?`
-		selectArgs = []any{payload.BundleIdentifier}
+		// match by bundle identifier first, or standard matching if we don't have a bundle identifier match
+		selectStmt = `SELECT id FROM software_titles WHERE bundle_identifier = ? OR (name = ? AND source = ? AND browser = '') ORDER BY bundle_identifier = ? DESC LIMIT 1`
```

Comment:
```suggestion
		selectStmt = `SELECT id FROM software_titles WHERE bundle_identifier = ? OR (name = ? AND source = ? AND browser = '' and bundle_identifier = '') ORDER BY bundle_identifier = ? DESC LIMIT 1`
```
Maybe on the `OR (name ...` also explicitly check that the `bundle_identifier` is empty too?

Reply:
If we forced empty bundle ID (would actually be an IS NULL check), we'd still be bugged for cases where the bundle ID is different but the source + name are the same, because source + name are where the uniqueness constraint is, so inserts would fail and we'd get no rows again for that path. Basically we're using bundle ID as more of a nice-to-have here, which from what I can tell is fine, particularly for this rev.